File: Day<07>/Solution.py
from bisect import bisect_left as lower_bound 
class Solution:
    def ValidPair(self, a, n): 
    	a = sorted(a) 
    	ans = 0
    	for i in range(n): 
    		if (a[i] <= 0): 
    			continue
    		# search for first element >= (-a[i] + 1)
    		j = lower_bound(a,-a[i] + 1) 
    		ans += i - j 
    	return ans

#{ 
#Driver Code Starts.

if __name__ == '__main__': 
	t = int(input())
	for _ in range(t):
		n = int(input())
		array = list(map(int, input().strip().split()))
		obj = Solution()
		ans = obj.ValidPair(array, n)
		print(ans)

File: Day<07>/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
    public:
    long long ValidPair(int a[], int n)
    {
    	sort(a, a+n);
    	long long ans = 0;
    	for (int i = 0; i < n; ++i)
		{
	    	if (a[i] <= 0)
    			continue;
    	    // search for first element >= (-a[i] + 1)
    	    int j = lower_bound(a, a + n, -a[i] + 1) - a;
    	    ans += (long long )(i - j);
    	}
    	return ans;
    }
};

int main()
{
    int t;
    cin>>t;
    while (t--)
    {
		int n;
		cin>>n;
		int array[n];
		for (int i = 0; i < n; ++i)
	    	cin>>array[i];
	 	Solution obj;
	 	cout << obj.ValidPair(array, n) <<"\n";
    }
    return 0; 
}
